packages/astro/src/default/components/OpenInStackblitzLink.astro

```diff
@@ -23,71 +23,17 @@
 
   function onClick() {
     const lesson = tutorialStore.lesson;
+    const snapshot = tutorialStore.takeSnapshot();
 
     if (!lesson) {
       throw new Error('Missing lesson');
     }
 
-    const files: Record<string, string> = {};
-
-    // first add template files
-    for (const [filePath, value] of Object.entries(tutorialStore.template || {})) {
-      files[removeLeadingSlash(filePath)] = value.toString();
-    }
-
-    // next overwrite with files from editor
-    for (const { filePath, value } of Object.values(tutorialStore.documents.get())) {
-      files[removeLeadingSlash(filePath)] = value.toString();
-    }
-
-    const packageJson = parseJson(files['package.json']);
-
-    // add start commands
-    if (files['package.json']) {
-      const mainCommand = resolveCommand(lesson.data.mainCommand);
-      const prepareCommands = (lesson.data.prepareCommands || []).map(resolveCommand);
-      const startCommand = [...prepareCommands, mainCommand].filter(Boolean).join(' && ');
-
-      files['package.json'] = JSON.stringify({ ...packageJson, stackblitz: { startCommand } }, null, 2);
-    }
-
     StackBlitzSDK.openProject({
       title: lesson.data.title || 'Lesson',
```

Comment:
I'm starting to think that maybe this should instead use the title of the tutorial instead of the title of the lesson.

Even if that means the same name for every lesson I think this will avoid strange names.

What do you think?

Reply:
What's title of the tutorial? Setting `title` in frontmatter for `type: tutorial` is not valid. Following configuration will throw validation error:

```md
---
type: tutorial
title: Example
---
```

Reply:
So the idea is to add `title` as an optional property of `tutorialSchema` in `types/src/schemas/tutorial.ts`. That way we don't break existing users and don't force them to have one.

For the default value here I would set something like `Project generated by TutorialKit` instead of `Lesson`.

Reply:
> So the idea is to add title as an optional property of `tutorialSchema` in `types/src/schemas/tutorial.ts`.

Do we want to show this somewhere on the UI as well? Not sure if this is getting out-of-scope regarding this PR. 🤔

Reply:
> Do we want to show this somewhere on the UI as well?

I don't think we want to. But now that you've said that I don't think my proposition makes much sense actually.

What if instead we used a different name of the property `openInStackBlitzLink` to be `openInStackBlitz`:

```ts
openInStackBlitz: z.union([
  z.boolean(), // to have it allowed or disallowed
  z.strictObject({
    projectTitle: z.string().optional(),
    projectDescription: z.string().optional(),
  }),
])
```

I'm also fine with not using anything and hardcoding `Project generated by TutorialKit` for now. I just think that people will probably want to tweak this at some point, so if we could add it here the better.

Reply:
The object-type config is great idea! I've implemented that now. In future we might want to expose options like `template` there, as we currently hardcode `template: 'node'` there.

Reply:
Oh that's a really good point!! 🤩  I feel like we should add it too, what do you think? 😃

Reply:
Added `projectTemplate` now.